ChiTech/ChiMesh/UnpartitionedMesh/unpartmesh_01_readfromvtu.cc

```diff
@@ -300,25 +361,94 @@ void chi_mesh::UnpartitionedMesh::
     << total_cell_count << " "
     << total_point_count;
 
-  //======================================== Lambda's
-  auto CellIs3D = [](int vtk_celltype)
+  //======================================== Determine if reading
+  //                                         cell identifiers
+  if (options.material_id_fieldname != options.boundary_id_fieldname)
   {
-    if (vtk_celltype == VTK_POLYHEDRON or
-        vtk_celltype == VTK_HEXAHEDRON or
-        vtk_celltype == VTK_TETRA)
-      return true;
-
-    return false;
-  };
-
-  //======================================== Scan cells for 3D
-  bool mesh_is_2D = true;
-  for (int c=0; c<total_cell_count; ++c)
-    if (CellIs3D(ugrid->GetCell(c)->GetCellType()))
+    chi_log.Log(LOG_ALLERROR)
+      << "The VTU reader expects material identifiers and boundary identifiers "
+      << "to be defined in the same field.";
+    std::exit(EXIT_FAILURE);
+  }
+
+  vtkDataArray* cell_id_array_ptr = nullptr;
+  if (options.material_id_fieldname.size() == 0)
```

Comment:
Clang-tidy indicates to use `options.material_id_fieldname.empty()` instead of `size()==0`.

Reply:
Definition in libc++ (https://github.com/llvm/llvm-project/blob/main/libcxx/include/string#L984):
```
    bool empty() const _NOEXCEPT {return size() == 0;}
```
Likewise, definition in libstdc++ is equivalent.